src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -917,6 +938,8 @@ public Solution WithDocumentFolders(DocumentId documentId, IEnumerable<string> f
         /// <summary>
         /// Creates a new solution instance with the document specified updated to have the specified file path.
         /// </summary>
+        // TODO: SolutionState.WithDocumentFilePath will throw if filePath is null, but it's odd because we
```

Comment:
Can we have an issue to track this? This looks like an API flaw so it's worth documenting outside of a `TODO`